pace-util/tests/test_zarr_monitor.py

```diff
@@ -394,3 +398,72 @@ def test_monitor_file_store_inconsistent_calendars(
         monitor.store(initial_state)
         with pytest.raises(ValueError, match="Calendar type"):
             monitor.store(final_state)
+
+
+@pytest.fixture(
+    params=[
+        ["x", "y"],
+        ["x", "y", "z"],
+        ["x_interface", "y", "z"],
+        ["x", "y_interface", "z"],
+        ["x", "y", "z_soil"],
+    ],
+)
+def diag(request, numpy):
+    dims = request.param
+    diag = pace.util.Quantity(
+        numpy.ones([size + 2 for size in range(len(dims))]), dims=dims, units="m"
+    )
+    return diag
+
+
+def _transpose(quantity, dims_2d=[Y_DIMS, X_DIMS], dims_3d=[Z_DIMS, Y_DIMS, X_DIMS]):
+    if len(quantity.dims) == 2:
+        return quantity.transpose(dims_2d)
+    elif len(quantity.dims) == 3:
+        return quantity.transpose(dims_3d)
+
+
+@requires_zarr
+def test_transposed_diags_write_across_ranks(diag, cube_partitioner, tmpdir_factory):
+
+    layout = (1, 1)
+    total_ranks = 6 * layout[0] * layout[1]
+    tmpdir = tmpdir_factory.mktemp("data.zarr")
+    store = zarr.storage.DirectoryStore(tmpdir)
+    shared_buffer = {}
+    for rank in range(total_ranks):
+        monitor = pace.util.ZarrMonitor(
+            store,
+            cube_partitioner,
+            mpi_comm=DummyComm(
+                rank=rank, total_ranks=total_ranks, buffer_dict=shared_buffer
+            ),
+        )
+        if rank % 2 == 0:
+            diag_to_store = _transpose(diag)
+        else:
+            diag_to_store = _transpose(
+                diag, dims_2d=[X_DIMS, Y_DIMS], dims_3d=[X_DIMS, Y_DIMS, Z_DIMS]
+            )
```

Comment:
Could you please make the arguments required and move the currently-default arguments here and into the other test to be more explicit (at the cost of repetition)? It would make it clearer to be able to read that the dimension transposing is different each time, also that they do get transposed in both cases (as opposed to `_transpose` doing nothing if passed no args).